mlos_core/mlos_core/optimizers/optimizer.py

```diff
@@ -26,7 +26,7 @@ class BaseOptimizer(metaclass=ABCMeta):
 
     def __init__(self, *,
                  parameter_space: ConfigSpace.ConfigurationSpace,
-                 optimization_targets: List[str],
+                 optimization_targets: str | List[str] | None = None,
```

Comment:
```suggestion
                 optimization_targets: Optional[Union[str, List[str]] = None,
```

We still need to support python 3.8